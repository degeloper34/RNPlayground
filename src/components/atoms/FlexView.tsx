import {View} from "react-native";

interface IProps {
  flex?: number;
  children?: React.ReactNode;
}

export function FlexView({flex = 1, children}: IProps) {
  return (
    <View
      style={{
        flex: flex || 1,
      }}
    >
      {children}
    </View>
  );
}
