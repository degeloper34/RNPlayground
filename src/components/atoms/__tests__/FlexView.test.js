import renderer from "react-test-renderer";
import React from "react";
import {FlexView} from "../FlexView";

describe("FlexView", () => {
  it("renders correctly without flex", () => {
    const component = renderer.create(<FlexView />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with flex 1", () => {
    const component = renderer.create(<FlexView flex={1} />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with flex 2", () => {
    const component = renderer.create(<FlexView flex={2} />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
