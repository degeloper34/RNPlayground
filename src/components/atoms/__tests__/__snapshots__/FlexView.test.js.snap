// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexView renders correctly with flex 1 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`FlexView renders correctly with flex 2 1`] = `
<View
  style={
    Object {
      "flex": 2,
    }
  }
/>
`;

exports[`FlexView renders correctly without flex 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
